commons/all/src/main/java/org/infinispan/commons/util/IntSets.java

```diff
@@ -189,6 +190,10 @@ public static IntSet concurrentSet(int maxExclusive) {
     * @return concurrent copy
     */
    public static IntSet concurrentCopyFrom(IntSet intSet, int maxExclusive) {
+      // if maxExclusive = 0; then we have an empty set
+      if (maxExclusive < 1) {
```

Comment:
I would personally say only return immutable empty set when  maxExclusive == 0. That way we will get the new exception you added above if it is negative.